src/trackedit/internal/au3/au3interaction.cpp

```diff
@@ -1521,6 +1617,55 @@ bool Au3Interaction::trimClipRight(const ClipKey& clipKey, secs_t deltaSec, secs
     return true;
 }
 
+bool Au3Interaction::stretchClipLeft(const ClipKey& clipKey,
+                                     secs_t deltaSec,
+                                     secs_t minClipDuration,
+                                     bool completed)
+{
+    //! NOTE: other clips must follow if selected
+    ClipKeyList clips = determineClipsForInteraction(clipKey);
+
+    secs_t adjustedDelta = clampLeftStretchDelta(clips, deltaSec, minClipDuration);
+
+    //! NOTE: don't be tempted to early return if delta is 0.0 or by any other reason:
```

Comment:
I would suggest writing tests for this rather than adding this comment everywhere :)